base/libgit2/types.jl

```diff
@@ -705,13 +705,19 @@ function getobjecttype(obj_type::Cint)
     end
 end
 
+import Base.securezero!
+
 "Credentials that support only `user` and `password` parameters"
 type UserPasswordCredentials <: AbstractCredentials
     user::AbstractString
```

Comment:
In general, you want to convert all string data to `String` unless it's really long and transcoding it would take a significant amount of time and/or computation or you need it in some other encoding for processing. In this case the strings are short and we don't need any other encoding, so `String` is the right type.